internal/app/wwctl/container/exec/main.go

```diff
@@ -20,23 +24,22 @@ import (
 /*
 fork off a process with a new PID space
 */
-func runContainedCmd(args []string) error {
-	var err error
-	if tempDir == "" {
-		tempDir, err = os.MkdirTemp(os.TempDir(), "overlay")
-		if err != nil {
-			wwlog.Warn("couldn't create temp dir for overlay", err)
-		}
-		defer func() {
-			err = os.RemoveAll(tempDir)
-			if err != nil {
-				wwlog.Warn("Couldn't remove temp dir for ephermal mounts:", err)
-			}
-		}()
+func runContainedCmd(args []string) (err error) {
+	conf := warewulfconf.Get()
+	if matches, _ := filepath.Glob(path.Join(conf.Paths.WWChrootdir, args[0], args[0]) + "-run-*"); len(matches) > 0 {
+		return fmt.Errorf("found lock directories for container: %v", matches)
+	}
+	overlayDir, err = os.MkdirTemp(path.Join(conf.Paths.WWChrootdir, args[0]), args[0]+"-run-")
```

Comment:
This isn't a true lock: there's a race condition between the checking of the existence of a directory for another process and the creation of one for this process.

If you want this to be a lock, don't use `os.MkdirTemp`--you only need that to prevent collisions with other processes. Since you intentionally want to create and detect collisions, just use `os.Mkdir` and use the error handling as the lock detection.